test/Tests.Infrastructure/TestBase.cs

```diff
@@ -205,26 +229,7 @@ TestCertificatesHolder Generate()
                 {
                     throw new CryptographicException($"Unable to load the test certificate for the machine '{Environment.MachineName}'. Log: {log}", e);
                 }
-
-                if (certBytes.Length == 0)
-                    throw new CryptographicException($"Test certificate length is 0 bytes. Machine: '{Environment.MachineName}', Log: {log}");
-
-                string serverCertificatePath = null;
-                try
-                {
-                    serverCertificatePath = Path.GetTempFileName();
-                    File.WriteAllBytes(serverCertificatePath, certBytes);
-                }
-                catch (Exception e)
-                {
-                    throw new InvalidOperationException("Failed to write the test certificate to a temp file." +
-                                                        $"tempFileName = {serverCertificatePath}" +
-                                                        $"certBytes.Length = {certBytes.Length}" +
-                                                        $"MachineName = {Environment.MachineName}.", e);
-                }
-
-                GlobalPathsToDelete.Add(serverCertificatePath);
```

Comment:
we are not registering certificate for deletion?

Reply:
No, I do that intentionally. To speed up tests, we preserve the test certificate for 24 hours